src/opendkim.cc

```diff
@@ -811,6 +812,38 @@ NAN_METHOD(OpenDKIM::SetOption) {
   info.GetReturnValue().Set(info.This());
 }
 
+#define MAXHDRCOUNT 10000
+
+NAN_METHOD(OpenDKIM::OHDRS) {
+   OpenDKIM* obj = Nan::ObjectWrap::Unwrap<OpenDKIM>(info.Holder());
+   unsigned char *ohdrs[MAXHDRCOUNT];
+   int nhdrs = MAXHDRCOUNT;
+   DKIM_STAT statp = DKIM_STAT_OK;
+
+   Isolate* isolate = info.GetIsolate(); // Needed to make new array
+
+   if (obj->sig == NULL) {
+      Nan::ThrowTypeError("ohdrs(): either there was no signature or called before eom() or chunk_end()");
+      return;
+   }
+
+   statp = dkim_ohdrs(obj->dkim, obj->sig, ohdrs, &nhdrs);
+
+   if (statp != DKIM_STAT_OK) {
+      throw_error(statp);
+      return;
+   }
+
+   Local<Array> zheaders = Array::New(isolate);
+   for (int i = 0; i < nhdrs; i++) {
+      char dump[MAXHDRCOUNT];        // We need dump to transfer from unsigned char
+      sprintf(dump, "%s", ohdrs[i]);
+      Nan::Set(zheaders, i, Nan::New(dump).ToLocalChecked());
```

Comment:
any reason not to just do:
```
Local<Array> zheaders = Array::New(isolate);
for (int i = 0; i < nhdrs; i++) {
  Nan::Set(zheaders, i, Nan::New(ohdrs[i]).ToLocalChecked());
}
```
Since we are making a New() string here, we have that memory allocation and copy.  Unless there is a really good reason I'm missing, we don't also want to set an arbitrary header length limit, which is what `dump[MAXHDRCOUNT]` will do, not to mention the additional copy from `sprintf(dump, "%s", ohdrs[i]);`.  I see the comment about `We need dump to transfer from unsigned char`, but perhaps we can just cast out of unsigned char?  I'm not sure what you may have tried.

Reply:
A cast is necessary but my dump variable is not. Thought I had tried that, apparently not. Going to push a new commit.